### PR TITLE
#134 | Destroying and cancelling process is done

### PR DESCRIPTION
**Pull request by Omarkojak**
for issue #134

### CODE REVIEW

**Comment by Omarkojak** [comment]
@AhmadElsagheer

File: app/Http/Controllers/Event/EventController.php [hunk 1]
```diff
@@ -32,9 +32,11 @@ public function __construct()
         ]]);
 
         $this->middleware('auth_organization', ['only' => [
-            'create', 'store', 'edit', 'update', 'destroy',
+            'create', 'store', 'edit', 'update', 'organization_cancel'
```
Comment (AhmadElsagheer) [inline app/Http/Controllers/Event/EventController.php]: `cancel`

File: app/Http/Controllers/Event/EventController.php [hunk 2]
```diff
@@ -32,9 +32,11 @@ public function __construct()
...
         ]]);
-    }
+
+		$this->middleware('auth_admin', ['only' => ['admin_cancel', 'destroy']]);
```
Comment (AhmadElsagheer) [inline app/Http/Controllers/Event/EventController.php]: no `admin_cancel`

File: app/Http/Services/EventService.php [hunk 3]
```diff
@@ -51,18 +51,48 @@ public function update(EventRequest $request, $id)
...
+		}
+	}
+
+	/**
+	 * Delete an event permanently.
+	 */
+	public function destroy($id ,$user)
+	{
+		$event = Event::findOrFail($id);
+		if($user->role >= 8)
```
Comment (AhmadElsagheer) [inline app/Http/Services/EventService.php]: The middleware does the jon

File: app/Http/Controllers/Event/EventController.php (hunk 2 above)
Comment (Omarkojak) [inline app/Http/Controllers/Event/EventController.php]: enta eli oltly kda :D
hashelha msh moshkla

Comment (AhmadElsagheer) [inline app/Http/Controllers/Event/EventController.php]: I told you we have a cancel function for organizations and destroy function for admins

**Comment by Omarkojak** [comment]
@AhmadElsagheer

File: app/Http/Controllers/Event/EventController.php [hunk 4]
```diff
@@ -32,9 +32,11 @@ public function __construct()
         ]]);
 
         $this->middleware('auth_organization', ['only' => [
-            'create', 'store', 'edit', 'update', 'destroy',
+            'create', 'store', 'edit', 'update'
```
Comment (AhmadElsagheer) [inline app/Http/Controllers/Event/EventController.php:35]: add `cancel` here and remov checker in the function

File: app/Http/Services/EventService.php [hunk 5]
```diff
@@ -51,17 +52,26 @@ public function update(EventRequest $request, $id)
 	}
 
 	/**
-	 * Cancel an event.
+	 * Organization, Admin cancel an event.
```
Comment (AhmadElsagheer) [inline app/Http/Services/EventService.php:55]: Organization only

File: resources/views/home.blade.php [hunk 6]
```diff
@@ -47,5 +47,14 @@
             </div>
         </div>
     </div>
+    <div class="container">
```
Comment (AhmadElsagheer) [inline resources/views/home.blade.php:50]: This shouldn't be here

File: app/Http/Controllers/Event/EventController.php (hunk 4 above)
Comment (Omarkojak) [inline app/Http/Controllers/Event/EventController.php:35]: w hadd it f eladmin brdo ?

File: resources/views/event/show.blade.php [hunk 7]
```diff
@@ -15,13 +15,22 @@
            <h1>
                {{--  Cancel Event --}}
                @if ($creator || (Auth::user() && Auth::user()->role >= 8))
```
Comment (AhmadElsagheer) [inline resources/views/event/show.blade.php:17]: edit condition

File: app/Http/routes.php [hunk 8]
```diff
@@ -193,6 +193,11 @@
     Route::post('feedback' , 'Volunteer\VolunteerController@storeFeedback');
 
     /**
+     * Volunteer view organizations he subscribes to
+     */
+    Route::get('dashboard/subscribed_organizations', 'Volunteer\VolunteerController@showSubscribedOrganizations');
```
Comment (AhmadElsagheer) [inline app/Http/routes.php:198]: This shouldn't be here

File: app/Http/Services/EventService.php [hunk 9]
```diff
@@ -84,9 +94,25 @@ public function unfollow($id, $volunteer)
...
+	  [
+		'phone' => 'required',
+		'address' => 'required',
+		'city' => 'required'
+	  ]);
+	  $event = Event::findOrFail($id);
+	  if($validator->passes() || !($event->required_contact_info))
+	  {
+		if ($event->timing > carbon::now())
+			Auth::user()->registerEvent($id);
```
Comment (AhmadElsagheer) [inline app/Http/Services/EventService.php:113]: use service

**Comment by Omarkojak** [comment]
sagheer bor7a enta 2ayly en eladmin can cancel lma kona bnchat :D

File: app/Http/Services/EventService.php (hunk 9 above)
Comment (AhmadElsagheer) [inline app/Http/Services/EventService.php:113]: Contact info shouldn't be here as well

**Comment by AhmadElsagheer** [comment]
This pull request has been mixed with other pull requests. Please clean it.
The dscription of the issue is clear.

**Comment by Omarkojak** [comment]
tb habd2 mn elawl 5als hadelete elbranch w abd2 mn elawl

**Pull request closed without merging**